lwb/metalang/sdf3/sdf3/src/main/java/mb/sdf3/task/debug/Sdf3ShowSpecParenthesizer.java

```diff
@@ -74,20 +81,23 @@ public Args(ResourcePath root, boolean concrete) {
     }
 
     @Override public CommandFeedback exec(ExecContext context, Args args) throws Exception {
-        return context.require(configFunctionWrapper.get(), args.root).mapOrElse(
-            o -> o.mapOrElse(
-                c -> run(context, c, args),
-                () -> CommandFeedback.of(ShowFeedback.showText("Cannot show parenthesizer; SDF3 was not configured in '" + args.root + "'", getName(args.concrete, args.root)))
-            ),
-            // TODO: should we propagate configuration errors here? Every task that requires some configuration will
-            //       propagate the same configuration errors, which would lead to duplicates.
-            e -> CommandFeedback.ofTryExtractMessagesFrom(e, args.root)
-        );
+        try {
+            final Sdf3SpecConfig specConfig = context.require(specConfigFunctionWrapper.get(), args.root).unwrap().unwrap();
+            final Sdf3Config config = context.require(configSupplierWrapper.get()).unwrap().unwrap();
+
+            return run(context, specConfig, config, "" /* TODO: Set this correctly */, args);
+        } catch (NoSuchElementException ex) {
+            return CommandFeedback.of(ShowFeedback.showText("Cannot show parenthesizer; SDF3 was not configured in '" + args.root + "'", getName(args.concrete, args.root)));
+        } catch (Exception ex) {
+            return CommandFeedback.ofTryExtractMessagesFrom(ex, args.root);
```

Comment:
Catching all `Exception`s is bad practice, as it also catches all `RuntimeException`s indicating bugs such as `NullPointerException` which should just crash the build, as well as catching `InterruptedException`s which should cancel the build, etc.

Also, I don't think any code here is throwing `NoSuchElementException`. 

AFAICS the old code was handling the exceptions as part of results correctly, so why change it?

Reply:
`NoSuchElementException` is thrown by `Option.unwrap()`. The reason I wrote it like this is because I need both the `Sdf3SpecConfig` and the `Sdf3Config`. Using the syntax as you wrote it, it would become something with deep nesting and multiple places of error handing:
```java
return context.require(specConfigFunctionWrapper.get(), args.root).mapOrElse(
    specConfigOpt -> specConfigOpt.mapOrElse(
        specConfig -> context.require(configSupplierWrapper.get()).mapOrElse(
            configOpt -> configOpt.mapOrElse(
                config -> run(context, specConfig, config, "", args),
                () -> CommandFeedback.of(ShowFeedback.showText("Cannot show parenthesizer; SDF3 was not configured in '" + args.root + "'", getName(args.concrete, args.root)))
            ),
            // TODO: should we propagate configuration errors here? Every task that requires some configuration will
            //       propagate the same configuration errors, which would lead to duplicates.
            e -> CommandFeedback.ofTryExtractMessagesFrom(e, args.root)
        ),
        () -> CommandFeedback.of(ShowFeedback.showText("Cannot show parenthesizer; SDF3 was not configured in '" + args.root + "'", getName(args.concrete, args.root)))
    ),
    // TODO: should we propagate configuration errors here? Every task that requires some configuration will
    //       propagate the same configuration errors, which would lead to duplicates.
    e -> CommandFeedback.ofTryExtractMessagesFrom(e, args.root)
);
```
Or do you have a better suggestion?

Reply:
If throwing/catching leads to better code, that is fine, as long as you rethrow `RuntimeException` and `InterruptedException` when you catch all `Exception`.